sys/docker_build_alpine_image

```diff
@@ -0,0 +1,328 @@
+            echo "alias ls=\"\\ls -F\"" >>/home/$uname/.bashrc && \
+            echo "alias q=\"exit\"" >>/home/$uname/.bashrc \
+        ) \
+    ) && ( \
+        cd /mnt && \
+        git clone -b $r2branch --depth 1 $github && \
+        cd radare2 && \
+        git checkout $r2commit && \
+        ./sys/install.sh && \
+        make install \
+    ) && ( \
+        apk del --purge -r \
+            g++ \
+            gcc \
+            linux-headers \
+            make \
+            npm \
+            py2-pip && \
+        rm -rf /tmp/* /var/cache/apk/* /var/tmp/* \
+    )
+
+# Initialize env
+USER $(id -nu)
+WORKDIR /r2
+
+# Setup r2pm
+RUN set -o pipefail && \
+    r2pm init && \
+    r2pm update
+
+CMD ["/bin/bash"]
+EOF
+
+# Tag old images
+info "Tagging any old r2_alpine images"
+while read -r tag; do
+    case "$tag" in
+        "r2_alpine:"*) docker image tag "${tag#*:}" "$tag" ;;
+    esac
+done < <(docker images | awk '{print $1":"$3}'); unset tag
+
+base="alpine"
+findbase="$(
+    docker images | \
+    grep -E "^(docker\.io\/)?${base//\//\\\/} +latest "
+)"
+
+# Build image (may take a while)
+info "Building image..."
+info "This may take a long time..."
+
+# Pull newest base image and build r2_alpine image
+docker pull $base:latest
+(
+    cd $tmp_docker_dir || errx 3 "$tmp_docker_dir not found"
+    # shellcheck disable=SC2154
+    docker build \
+        ${http_proxy:+--build-arg http_proxy=$http_proxy} \
+        ${https_proxy:+--build-arg https_proxy=$https_proxy} \
+        -t r2_alpine:latest .
+)
+
+# Only remove base image if it didn't already exist
+[[ -n $findbase ]] || docker rmi $base:latest
+
+info "done"
+
+old_base="^(docker\.io\/)?${base//\//\\\/} +<none>"
+old_r2="^r2_alpine +[^l ]"
+found="$(docker images | grep -E "($old_base)|($old_r2)")"
+if [[ -n $found ]]; then
+    # List old images
+    echo
+    info "Old images:"
+    docker images | head -n 1
+
+    while read -r line; do
+        echo "$line"
+    done < <(
+        docker images | \
+        grep -E "($old_base)|($old_r2)"
+    ); unset line
+
+    # Prompt to remove old images
+    unset remove
+    echo
+    while :; do
+        read -n 1 -p "Remove old images (y/N/q)? " -rs ans
+        echo
+        case "$ans" in
+            ""|"n"|"N"|"q"|"Q") break ;;
+            "y"|"Y") remove="true"; break ;;
+            *) echo "Invalid choice" ;;
+        esac
+    done
+
+    if [[ -n $remove ]]; then
+        # Remove old images
+        while read -r tag; do
+            docker rmi "$tag"
+        done < <(
+            docker images | awk "/$old_r2/ {print \$1\":\"\$3}"
+        ); unset tag
+
+        while read -r id; do
+            docker rmi "$id"
+        done < <(
+            docker images | awk "/$old_base/ {print \$3}"
+        ); unset id
+    fi
+fi; unset found
+
+cleanup 0
+
+cat <<EOF
+
+It's suggested you add something like the following to your ~/.bashrc:
+
+r2d() {
+    local rm
```

Comment:
i prefer variablenames in uppercase in shellscripts

Reply:
I thought uppercase variables were for variables that get exported or inherited from the env. Is that not a standard? I'm pretty sure shellcheck complains about that.